pywatts/wrapper/sm_time_series_model_wrapper.py

```diff
@@ -0,0 +1,170 @@
+        :rtype: Dict
+        """
+        return {
+            "module_kwargs": self.module_kwargs,
+            "fit_kwargs": self.fit_kwargs,
+            "predict_kwargs": self.predict_kwargs,
+        }
+
+    def set_params(self, module_kwargs=None, fit_kwargs=None, predict_kwargs=None):
+        """
+        Set the parameters of the statsmodels wrapper
+
+        :param module_kwargs: keyword arguments for the statsmodel module.
+        :type module_kwargs: Dict
+        :param fit_kwargs: keyword arguments for the fit method.
+        :type fit_kwargs: Dict
+        :param predict_kwargs: keyword arguments for the predict method.
+        :type predict_kwargs: Dict
+        """
+        if module_kwargs:
+            self.module_kwargs = module_kwargs
+        if fit_kwargs:
+            self.fit_kwargs = fit_kwargs
+        if predict_kwargs:
+            self.predict_kwargs = predict_kwargs
+
+    def fit(self, **kwargs: xr.DataArray):
+        """
+        Fits the statsmodels module
+
+        :param kwargs: A dict of input arrays
+        :type kwargs: xr.DataArray
+        """
+        x = []
+        y = []
+        for key, value in kwargs.items():
+            if key.startswith("target"):
+                y.append(value.values.reshape(-1))
+            else:
+                x.append(value.values)
+
+        if "exog" in inspect.signature(self.module).parameters or "kwargs" in inspect.signature(
+                self.module).parameters:
+            self.model = self.module(endog=np.stack(y, axis=-1), exog=np.concatenate(x, axis=-1), **self.module_kwargs).fit(
+                **self.fit_kwargs)
+        else:
+            self.model = self.module(endog=np.stack(y, axis=-1), **self.module_kwargs).fit(**self.fit_kwargs)
+
+        self.is_fitted = True
+
+    def transform(self, **kwargs: xr.DataArray) -> xr.DataArray:
+        """
+        Predicts the result with the wrapped statsmodels module
+
+        :param kwargs: A dict of input arrays
+        :type kwargs: xr.DataArray
+        :return: the transformed dataarray
+        :rtype: xr.DataArray
+        """
+        time_data = list(kwargs.values())[0][_get_time_indeces(kwargs)[0]]
+
+        x = []
+        for key, value in kwargs.items():
+            x.append(value.values)
+
+        if hasattr(self.model, "forecast"):
+            if "exog" in inspect.signature(self.model.forecast).parameters or "kwargs" in inspect.signature(
+                    self.model.forecast).parameters:
+                prediction = self.model.forecast(len(time_data), exog=np.concatenate(x, axis=-1), **self.predict_kwargs)[0]
+
+            else:
+                prediction = self.model.forecast(len(time_data), **self.predict_kwargs)[0]
+        else:
+            raise Exception(f"{self.module.__class__.__name__} has not forecast method...")
+
+        return numpy_to_xarray(prediction, list(kwargs.values())[0], self.name)
+
+    def save(self, fm: FileManager):
```

Comment:
Does the save method also save module_kwargs, fit_kwargs, and predict_kwargs?

Reply:
Yes via calling save from super classes